src/build-adapter.ts

```diff
@@ -0,0 +1,133 @@
+import {
+	handleBuildAllCommand,
+	handleBuildReactCommand,
+	handleBuildTypeScriptCommand,
+	parseOptions,
+} from "./build-adapter-handlers";
+import { interceptErrors } from "./util";
+import yargs = require("yargs/yargs");
+
+const parser = yargs(process.argv.slice(2));
+parser
+	.env("IOBROKER_BUILD")
+	.strict()
+	.usage("ioBroker adapter build script\n\nUsage: $0 <command> [options]")
+	.alias("h", "help")
+	.alias("v", "version")
+	.command(
+		["react"],
+		"Compiles React sources",
+		{},
+		interceptErrors(handleBuildReactCommand),
+	)
+	.command(
+		["typescript", "ts"],
+		"Compile TypeScript sources",
+		{},
+		interceptErrors(handleBuildTypeScriptCommand),
+	)
+	.command(
+		["all"],
+		"Compile all of the above",
+		{},
+		interceptErrors(handleBuildAllCommand),
+	)
+	.options({
+		watch: {
+			type: "boolean",
+			default: false,
+			description: "Watch for changes and recompile",
+		},
+		reactRootDir: {
```

Comment:
Also, please provide short forms for all (or most) commands (e.g.  `-r` for root, `-o` for output, ....)

Reply:
AFAIK yargs adds kebab-case arguments by default.

> Also, please provide short forms for all (or most) commands (e.g. -r for root, -o for output, ....)

I thought about that, but found it unintuitive when switching from the individual commands (react, typescript) to the `all` command where we would need the same options but with different short aliases. As such, `build all -r /path` would have a different meaning than `build ts -r /path`.

Reply:
Since there are so many options, what do you think about providing a `.buildconfig.json` file (and an option to load this instead), similar to https://github.com/AlCalzone/release-script/blob/b06a628d878347f345404fb21c1fb29a1cfe2c2c/packages/release-script/src/index.ts#L166-L171
?

Reply:
Very good idea about the config file! Can you please implement it? :-)